src/App.tsx

```diff
@@ -15,48 +15,112 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+const SORT_FIELD_LIGHT = 'is-light';
+
+enum SortType {
+  Alphabetical = 'alpha',
+  Length = 'length',
+  None = '',
+}
+
+function getPreparedGoods(
+  { sortField, sortedLength, isReversed }: {
+    sortField: SortType,
+    sortedLength: SortType,
+    isReversed: boolean,
+  },
+): string[] {
+  let preparedGoods = [...goodsFromServer];
+
+  if (sortField === SortType.Alphabetical) {
+    preparedGoods = [...preparedGoods].sort(
+      (good1, good2) => good1.localeCompare(good2),
+    );
+  }
+
+  if (sortedLength === SortType.Length) {
+    preparedGoods = [...preparedGoods].sort(
+      (good1, good2) => good1.length - good2.length,
+    );
+  }
+
+  if (isReversed) {
+    preparedGoods = [...preparedGoods].reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState<SortType>(SortType.None);
+  const [sortedLength, setSortedLength] = useState<SortType>(SortType.None);
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+
+  const visibleGoods: string[] = getPreparedGoods(
+    { sortField, sortedLength, isReversed },
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={sortField === 'alpha' ? 'button is-info'
+            : `button is-info ${SORT_FIELD_LIGHT}`}
+          onClick={() => {
+            setSortField(SortType.Alphabetical);
+            setSortedLength(SortType.None);
+          }}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={sortedLength === 'length' ? 'button is-success'
+            : `button is-success ${SORT_FIELD_LIGHT}`}
+          onClick={() => {
+            setSortedLength(SortType.Length);
+            setSortField(SortType.None);
+          }}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={isReversed
+            ? 'button is-warning'
+            : `button is-warning ${SORT_FIELD_LIGHT}`}
+          onClick={() => setIsReversed(prev => !prev)}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(sortField || sortedLength || isReversed) && (
+          <button
+            type="button"
+            className={`button is-danger ${SORT_FIELD_LIGHT}`}
+            onClick={() => {
+              setSortField(SortType.None);
+              setSortedLength(SortType.None);
+              setIsReversed(false);
+            }}
```

Comment:
make handler in component and pass just reference here.
```
const handleReset = () => {
    setSortField(SortType.None);
    setIsReversed(false);
 }
```
```suggestion
            onClick={handleReset}
```
 same for reverse button.